kernelci/data/kernelci_api.py

```diff
@@ -83,6 +95,47 @@ def get_nodes_by_commit_hash(self, commit_hash):
     def get_node_from_event(self, event):
         return self.get_node(event.data['id'])
 
+    def pubsub_event_filter(self, sub_id, event):
+        """Filter Pub/Sub events
+
+        Filter received Pub/Sub event using provided filter dictionary.
+        Return True if the event matches with the filter, otherwise False.
+        """
+        event_filter_status = True
+        filters = self._filters.get(sub_id)
+        if not filters:
+            return event_filter_status
+        for key, value in filters.items():
+            if key not in event.keys():
+                continue
+            if isinstance(value, dict):
```

Comment:
This only looks for one sub-level in the object, which is fine with the current data schema we have. I think we can merge this code, but ideally a follow-up would be to make it truly recursive and try to match filers with any levels of nested dictionary values.

Reply:
Okay.